src/wh_client_she.c

```diff
@@ -0,0 +1,883 @@
+{
+    int ret;
+    whPacket* packet;
+    if (c == NULL)
+        return WH_ERROR_BADARGS;
+    packet = (whPacket*)wh_CommClient_GetDataPtr(c->comm);
+    /* send rnd req */
+    ret = wh_Client_SendRequest(c, WH_MESSAGE_GROUP_SHE, WH_SHE_RND,
+        WOLFHSM_PACKET_STUB_SIZE, (uint8_t*)packet);
+    return ret;
+}
+
+int wh_Client_SheRndResponse(whClientContext* c, uint8_t* out, uint32_t* outSz)
+{
+    int ret;
+    uint16_t group;
+    uint16_t action;
+    uint16_t dataSz;
+    whPacket* packet;
+    if (c == NULL || out == NULL || outSz == NULL ||*outSz < WOLFHSM_SHE_KEY_SZ)
+        return WH_ERROR_BADARGS;
+    packet = (whPacket*)wh_CommClient_GetDataPtr(c->comm);
+    ret = wh_Client_RecvResponse(c, &group, &action, &dataSz, (uint8_t*)packet);
+    if (ret == 0) {
+        if (packet->rc != WOLFHSM_SHE_ERC_NO_ERROR)
+            ret = packet->rc;
+        else {
+            memcpy(out, packet->sheRndRes.rnd, sizeof(packet->sheRndRes.rnd));
+            *outSz = sizeof(packet->sheRndRes.rnd);
+        }
+    }
+    return ret;
+}
+
+int wh_Client_SheRnd(whClientContext* c, uint8_t* out, uint32_t* outSz)
+{
+    int ret;
+    ret = wh_Client_SheRndRequest(c);
+    if (ret == 0) {
+        do {
+            ret = wh_Client_SheRndResponse(c, out, outSz);
+        } while (ret == WH_ERROR_NOTREADY);
+    }
+    return ret;
+}
+
+int wh_Client_SheExtendSeedRequest(whClientContext* c, uint8_t* entropy,
+    uint32_t entropySz)
+{
+    int ret;
+    whPacket* packet;
+    if (c == NULL || entropy == NULL || entropySz != WOLFHSM_SHE_KEY_SZ)
+        return WH_ERROR_BADARGS;
+    packet = (whPacket*)wh_CommClient_GetDataPtr(c->comm);
+    /* set entropy */
+    memcpy(packet->sheExtendSeedReq.entropy, entropy,
+        sizeof(packet->sheExtendSeedReq.entropy));
+    /* send init rng req */
+    ret = wh_Client_SendRequest(c, WH_MESSAGE_GROUP_SHE, WH_SHE_EXTEND_SEED,
+        WOLFHSM_PACKET_STUB_SIZE, (uint8_t*)packet);
+    return ret;
+}
+
+int wh_Client_SheExtendSeedResponse(whClientContext* c)
+{
+    int ret;
+    uint16_t group;
+    uint16_t action;
+    uint16_t dataSz;
+    whPacket* packet;
+    if (c == NULL)
+        return WH_ERROR_BADARGS;
+    packet = (whPacket*)wh_CommClient_GetDataPtr(c->comm);
+    ret = wh_Client_RecvResponse(c, &group, &action, &dataSz, (uint8_t*)packet);
+    if (ret == 0)
+        ret = packet->rc;
+    return ret;
+}
+
+int wh_Client_SheExtendSeed(whClientContext* c, uint8_t* entropy,
+    uint32_t entropySz)
+{
+    int ret;
+    ret = wh_Client_SheExtendSeedRequest(c, entropy, entropySz);
+    if (ret == 0) {
+        do {
+            ret = wh_Client_SheExtendSeedResponse(c);
+        } while (ret == WH_ERROR_NOTREADY);
+    }
+    return ret;
+}
+
+int wh_Client_SheEncEcbRequest(whClientContext* c, uint8_t keyId, uint8_t* in,
+    uint32_t sz)
+{
+    int ret;
+    uint8_t* packIn;
+    whPacket* packet;
+    if (c == NULL || in == NULL || sz < WOLFHSM_SHE_KEY_SZ)
```

Comment:
Better error out if it can't be placed in a single packet.
```suggestion
    if (c == NULL || in == NULL || sz < WOLFHSM_SHE_KEY_SZ || sz > (WH_COMM_DATA_LEN - WOLFHSM_PACKET_STUB_SIZE - sizeof(packet->sheEncEcbReq))
```